src/Microsoft.AspNetCore.Http.Extensions/UriHelper.cs

```diff
@@ -101,7 +101,7 @@ public static string Encode(Uri uri)
         /// <returns></returns>
         public static string GetEncodedUrl(this HttpRequest request)
```

Comment:
Not sure I follow. Wouldn't it just be:

```C#
Headers.Set(HeaderNames.Location, value == null ? null : value.GetEncodedUrl());
```

Value's a Uri which is the contextual part. It's odd having this method and the below extension method. They both should be consistent whatever is decided.